### PR TITLE
Added solution for Project Euler problem 085

### PR DESCRIPTION
**Pull request by fpringle**
### **Describe your change:**

Project euler problem 85: [Counting Rectangles](https://projecteuler.net/problem=85)

By counting carefully it can be seen that a rectangular grid measuring 3 by 2 contains eighteen rectangles.

Although there exists no rectangular grid that contains exactly two million rectangles, find the area of the grid with the nearest solution.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_085/sol1.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        triangle_numbers.append(triangle_numbers[-1] + idx)
+
+    best_product: int = 0  # we want this to be as close as possible to target
+    area: int = 0  # the area a*b corresponding to the grid that gives
+    # the product closest to target
+    b_estimate: float  # an estimate of b, using the quadratic formula
+    b_floor: int  # the largest integer less than b_estimate
+    b_ceil: int  # the largest integer less than b_estimate
+    triangle_b_first_guess: int  # the triangle number corresponding to b_floor
+    triangle_b_second_guess: int  # the triangle number corresponding to b_ceil
```
Comment (dhruvmanila) [inline project_euler/problem_085/sol1.py:77]: One line comments should always be on the top. If there are only a few words, then they can be on the same line. This is done to provide better readability :)

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Minor tweaks and we're good to go!

File: project_euler/problem_085/sol1.py (hunk 1 above)
Comment (fpringle) [reply project_euler/problem_085/sol1.py]: Is that better?

Comment (dhruvmanila) [reply project_euler/problem_085/sol1.py]: Oh, I am sorry to be ambiguous but I meant on top of the variable itself instead of on the side.

**Review by dhruvmanila — APPROVED** [review]
LGTM! 🎉

**Pull request merged**